controller/resource_test.go

```diff
@@ -243,7 +245,6 @@ func TestTaskStateListerList(t *testing.T) {
 			resources, err := lister.List()
 			require.NoError(t, err)
 
-			sortTaskStates(c.expResources)
```

Comment:
This one uses `t.Parallel()`. So multiple test case were concurrently sorting `c.expResources`, which shares some data across test cases. I move the sort prior to the `t.Run`, so that the sorts don't occur in parallel since the subtest functions are not started until later (I think. This maybe could still be a race, but the race detector seems happy)